src/components/SetupTab/index.jsx

```diff
@@ -65,30 +377,31 @@ class ArgsForm extends React.Component {
 
   constructor(props) {
     super(props);
+
     this.handleChange = this.handleChange.bind(this);
+    this.handleBoolChange = this.handleBoolChange.bind(this);
     this.selectFile = this.selectFile.bind(this);
     this.onDragDrop = this.onDragDrop.bind(this);
   }
 
-  componentDidMount() {
-    // Validate args immediately on mount. Values will be `undefined`
-    // but converted to empty strings by `argsValuesFromSpec` 
-    // Validation will handle updating the `valid` property
-    // of the optional and conditionally required args so that they 
-    // can validate without any user-interaction. Validation messages
-    // won't appear to the user until an argument has been touched.
-
-    // TODO: could call batchUpdateArgs here instead
-    // to avoid passing investValidate to this component at all.
-    // this.props.batchUpdateArgs(JSON.parse(args_dict_string));
-    this.props.investValidate(argsValuesFromSpec(this.props.args));
+  handleChange(event) {
+    /** Pass input value up to InvestJob for storage & validation.
+    *
+    * For text fields, this is on the onKeyUp handler, specifically 
+    * so that we don't call investValidate over and over while 
+    * laying on the backspace key (or any key).
+    */
+    const value = event.target.value;
+    const argkey = event.target.name;
+    this.props.updateArgValues(argkey, value)

```

Comment:
Ah, I was wondering how we'd be able to propagate changes back to a higher level!  Makes sense.